src/webview/components/memory-table.tsx

```diff
@@ -329,9 +370,88 @@ export class MemoryTable extends React.PureComponent<MemoryTableProps, MemoryTab
+
+    protected determineGroupsPerRowToRender(): number {
+        const options = MemorySizeOptions.create(this.props, this.state);
+
+        if (this.props.groupsPerRow === 'Autofit') {
+            const container = this.datatableRef.current?.getElement().querySelector<HTMLElement>(`td.${DataColumn.CLASS_NAME}`);
+            if (container) {
+                return MemoryTable.approximateGroupsPerRow(container, options);
+            }
+            return 1;
+        }
+
+        return options.groupsPerRow;
+    }
 }
 
 export namespace MemoryTable {
     export const TABLE_CLASS = 'memory-inspector-table';
     export const GROUP_SEPARATOR = 'group-separator';
+
+    /**
+     * The approximation is done by:
+     *
+     * 1. Calculating the number of characters that would fit into the element
+     * 2. Calculating the number of characters that we require
+     * 3. Dividing the available characters by the characters we require
+     *    to aquire the groups per row we theoretically could fit into the element
+     * 4. Including other factors such as margins
+     *
+     * **Note**
+     *
+     * The returned `groups per row` value is based on {@link CONFIG_GROUPS_PER_ROW_NUMERIC_CHOICES}
+     *
+     * @returns Approximated groups per row that will fit into the element
+     */
+    export function approximateGroupsPerRow(element: HTMLElement, options: MemorySizeOptions): number {
+        const choices: number[] = [...CONFIG_GROUPS_PER_ROW_NUMERIC_CHOICES];
+
+        const columnWidth = elementInnerWidth(element);
+        const characterWidth = characterWidthInContainer(element, '0');
+        // Number of characters that fit into the element
+        const fittingCharacters = columnWidth / characterWidth;
+
+        // Characters that are rendered for each eight bits, e.g., <0F> = 2 Characters
+        // HTML: span.eight-bits
+        const eightBitCharacters = 2;
+        // Characters that are rendered for each single word, e.g., <0F><0F> = 4 Characters, 2 bytes per word
+        // HTML: span.single-word
+        const singleWordCharacters = eightBitCharacters * options.bytesPerWord;
+        // Characters that are rendered for each words per group, e.g., <0F0F> <0000> = 8 Characters, 2 words per group
+        // HTML: span.byte-group
+        const wordsPerGroupCharacters = singleWordCharacters * options.wordsPerGroup;
+        // Number of groups per row that would theoretically fit
+        const groupsPerRowFit = Math.ceil(fittingCharacters / wordsPerGroupCharacters);
+        // Find the nearest lowest possible choice
+        const groupsPerRowIndex = choices.reduce((prev, curr, index) => groupsPerRowFit >= curr ? index : prev, 0);
+        const groupsPerRow = choices[groupsPerRowIndex];
+
+        if (groupsPerRowIndex > 0) {
+            // Include other necessary calculations such as margins
+            // We can not look that up
+            const marginRights = (groupsPerRow - 1) * fontSize(element) * 0.2;
```

Comment:
Maybe add a comment to the CSS file where we define the right margin and vice versa?